packages/core/__tests__/language-server/folding-ranges.test.ts

```diff
@@ -0,0 +1,238 @@
+import { Project } from 'glint-monorepo-test-utils';
+import { describe, beforeEach, afterEach, test, expect } from 'vitest';
+import { stripIndent } from 'common-tags';
+
+describe('Language Server: Folding Ranges', () => {
+  let project!: Project;
+
+  beforeEach(async () => {
+    project = await Project.create();
+  });
+
+  afterEach(async () => {
+    await project.destroy();
+  });
+
+  test('function', () => {
+    project.write({
+      'example.ts': stripIndent`
+        function foo() {
+          return 'bar';
+        }
+      `,
+    });
+
+    let server = project.startLanguageServer();
+    let folds = server.getFoldingRanges(project.fileURI('example.ts'));
+
+    expect(folds).toEqual([
+      {
+        startLine: 0,
+        endLine: 1,
+        kind: undefined,
+      },
+    ]);
+  });
+
+  test('nested function', () => {
+    project.write({
+      'example.ts': stripIndent`
+        function topLevel() {
+
+          function nested() {
+            return 'bar';
+          }
+
+          return nested();
+        }
+      `,
+    });
+
+    let server = project.startLanguageServer();
+    let folds = server.getFoldingRanges(project.fileURI('example.ts'));
+
+    expect(folds).toEqual([
+      {
+        startLine: 0,
+        endLine: 6,
```

Comment:
is this excluding the last `}`?

Reply:
The last line is excluded if it's just a closing bracket (https://github.com/typed-ember/glint/pull/632/files#diff-cadffc59ac4fd3c9fd1cd7488980fc9cfea40cd5a93129f1021c54c810980e20R503). This is so that folded functions keep the `}` visible:
```ts
// Like this when folded
function topLevel() {
}

// instead of this
function topLevel() {
```

this is consistent with the TS language server's folding.